prow/plugins/updateconfig/updateconfig.go

```diff
@@ -65,55 +66,62 @@ type kubeClient interface {
 }
 
 func handlePullRequest(pc plugins.PluginClient, pre github.PullRequestEvent) error {
-	configFile := pc.PluginConfig.ConfigUpdater.ConfigFile
-	pluginFile := pc.PluginConfig.ConfigUpdater.PluginFile
-	return handle(pc.GitHubClient, pc.KubeClient, pc.Logger, pre, configFile, pluginFile)
+	return handle(pc.GitHubClient, pc.KubeClient, pc.Logger, pre, maps(pc))
 }
 
-func handleConfig(gc githubClient, kc kubeClient, org, repo, commit, configFile string) error {
-	content, err := gc.GetFile(org, repo, configFile, commit)
-	if err != nil {
-		return err
+func maps(pc plugins.PluginClient) map[string]string {
+	log := pc.Logger
+	cu := pc.PluginConfig.ConfigUpdater
+	m := map[string]string{}
+	if cu.ConfigFile != "" {
+		log.Warnf(`config_file is deprecated, please switch to "maps": {"%s": "config"} before July 2018`, cu.ConfigFile)
+		m[cu.ConfigFile] = "config"
 	}
-
-	c := kube.ConfigMap{
-		ObjectMeta: kube.ObjectMeta{
-			Name: "config",
-		},
-		Data: map[string]string{
-			"config": string(content),
-		},
+	if cu.PluginFile != "" {
+		log.Warnf(`plugin_file is deprecated, please switch to "maps": {"%s": "plugins"} before July 2018`, cu.PluginFile)
+		m[cu.PluginFile] = "plugins"
 	}
-
-	_, err = kc.ReplaceConfigMap("config", c)
-	return err
+	for f, n := range cu.Maps {
+		m[f] = n
+	}
+	return m
 }
 
-func handlePlugin(gc githubClient, kc kubeClient, org, repo, commit, pluginFile string) error {
-	content, err := gc.GetFile(org, repo, pluginFile, commit)
+func update(gc githubClient, kc kubeClient, org, repo, commit, filename, name string) error {
+	content, err := gc.GetFile(org, repo, filename, commit)
 	if err != nil {
-		return err
+		return fmt.Errorf("get file err: %v", err)
 	}
 
-	c := kube.ConfigMap{
+	c := string(content)
+
+	cm := kube.ConfigMap{
 		ObjectMeta: kube.ObjectMeta{
-			Name: "plugins",
+			Name: name,
 		},
 		Data: map[string]string{
-			"plugins": string(content),
+			name:                c,
```

Comment:
Why add the content twice?

Reply:
IMO when I say `{"prow/whatever.yaml": "plugins"}` and then map in the plugins configmap to `/etc/plugin`, it makes more sense to me to look for a `/etc/plugins/whatever.yaml` file than `/etc/plugins/plugins`.

I would like to eventually switch over our configs to use `/etc/plugins/plugins.yaml` instead.

I had to debug this situation on another team recently.

Thoughts/concerns?

Reply:
Ah, now I see. lgtm

Reply:
Cool

Reply:
Do we still plan to migrate this?